web/src/message_util.ts

```diff
@@ -84,6 +84,15 @@ export function add_new_messages_data(
     return msg_list_data.add_messages(messages);
 }
 
+export function get_count_of_messages_in_topic_sent_after_current_message(
+    stream_id: number,
+    topic: string,
+    message_id: number,
+): number {
+    const all_messages = get_messages_in_topic(stream_id, topic);
+    return all_messages.filter((msg) => msg.id >= message_id).length;
+}
```

Comment:
As noted in the chat.zulip.org discussion, this function (and its parallel code path) needs to be tweaked to also check if the message is already marked unread.

Reply:
@Tim Abbott, I’m a bit confused. Did you mean "already marked unread" or "already marked read"?

Reply:
Hey @timabbott , I am a bit confused in this portion. Could you explain me why we need to check if the message is already marked unread,

Reply:
Oh, never mind, this is not about "mark as unread". Yes this is fine.